pkg/controller/migplan/migplan_controller.go

```diff
@@ -445,3 +467,62 @@ func (r ReconcileMigPlan) deleteImageRegistryDeploymentForClient(client k8sclien
 	}
 	return nil
 }
+
+func (r ReconcileMigPlan) validateMigAnalytics(plan *migapi.MigPlan) error {
+	log.Info(fmt.Sprintf("Checking if migpanalytics exists or not"))
+	migAnalytics := &migapi.MigAnalyticList{}
+	err := r.List(context.TODO(), k8sclient.MatchingLabels(map[string]string{"migplan": plan.Name}), migAnalytics)
+	if err != nil {
+		if !errors.IsNotFound(err){
+			return err
+		}
+	}
+	for _, migAnalytic := range migAnalytics.Items {
+		if migAnalytic.Spec.AnalyzeExntendedPVCapacity && plan.Spec.Refresh{
+			log.Info("refreshing miganalytics")
+			migAnalytic.Spec.Refresh = true
+			err := r.Update(context.TODO(), &migAnalytic)
+			if err != nil{
+				return err
+			}
+			return nil
+		}
+	}
+
+	log.Info(fmt.Sprintf("Creating new analytics"))
+	pvMigAnalytics := &migapi.MigAnalytic{}
+	pvMigAnalytics.GenerateName = plan.Name + "-"
+	pvMigAnalytics.Namespace = plan.Namespace
```

Comment:
If we are listing MigAnalytics by labels, we need to make sure we add those labels on the new ones we create here

Reply:
The labels are added by the `MigAnalytic` controller I think. I ran this code and labels were present in the yaml of the resource.

Reply:
@JaydipGabani best to make sure migplan adds them as there is no coupling between the two, duplicate labels get overwritten anyway

Reply:
Added label for migplan